core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -136,7 +136,11 @@ wasi_addr_to_sockaddr(const __wasi_addr_t *wasi_addr,
 int
 accept(int sockfd, struct sockaddr *addr, socklen_t *addrlen)
 {
+#ifdef __cplusplus
+    __wasi_addr_t wasi_addr {} ;
+#else
     __wasi_addr_t wasi_addr = { 0 };
+#endif
```

Comment:
@dzobbe Shall we use memset to initialize it here so as to avoid introducing `__cplusplus` here?
```C
    __wasi_addr_t wasi_addr;
    ...
    memset(&wasi_addr, 0, sizeof(wasi_addr));
```